### PR TITLE
fix(Core\Scripts):Jiklik's bats from phase 2 now won't engage players…

### PR DESCRIPTION
**Pull request by Ragelezz**
… directly and will fly

## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/834

## Tests Performed:

- Engage Jiklik and change to phase 2 (50% of her total health) and see that the bats spawned during this phase will come down to players and meele them
- 

## How to Test the Changes:

1. Engage Jiklik and change to phase 2 (50% of her total health) and see that now the bats spawned during this phase will keep flying and throw missiles accordingly.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
build failing

**Comment by Ragelezz** [comment]
can't figure from what since its configure os which fails, any clues? i know something related to this https://github.com/azerothcore/azerothcore-wotlk/issues/8083 but its beyond me.

**Comment by Nyeriah** [comment]
@Ragelezz I ran the checks again and they passed this time, I don't think that failure was related to your changes, looks like some internal stuff that didn't happen again.

File: data/sql/updates/pending_db_world/rev_1632863354170415400.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1632863354170415400');
+
+
+UPDATE `creature_template` SET `InhabitType`='4' WHERE  `entry`=14965;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1632863354170415400.sql:4]: dont use singlequotes on non-string values

File: data/sql/updates/pending_db_world/rev_1632863354170415400.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+UPDATE `creature_template` SET `ScriptName`='npc_batrider' WHERE entry=14965;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1632863354170415400.sql:5]: please add backticks to `entry`

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp [hunk 3]
```diff
@@ -221,16 +222,17 @@ class npc_batrider : public CreatureScript
...
-                if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
-                {
-                    DoCast(target, SPELL_BOMB);
-                    Bomb_Timer = 5000;
-                }
+                std::list<Unit*> targets;
+                SelectTargetList(targets, 1, SELECT_TARGET_RANDOM, 500.0f, true);
+                if (!targets.empty())
+                    if (targets.size() > 1)
+                        targets.resize(1);
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp:229]: please add brackets

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp [hunk 4]
```diff
@@ -221,16 +222,17 @@ class npc_batrider : public CreatureScript
...
+
+                for (std::list<Unit*>::iterator itr = targets.begin(); itr != targets.end(); ++itr) me->CastSpell((*itr), SPELL_BOMB);
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp:231]:
```suggestion
                for (std::list<Unit*>::iterator itr = targets.begin(); itr != targets.end(); ++itr)
                {
                    me->CastSpell((*itr), SPELL_BOMB);
                }

```

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1632863354170415400.sql [hunk 5]
```diff
@@ -0,0 +1,9 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1632863354170415400');
+
+
+
+
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/rev_1632863354170415400.sql:5]:
```suggestion
```

**Comment by Ragelezz** [comment]
@azerothcore/testers

**Comment by Ragelezz** [comment]
This was already tested by me when I first made this PR, shall I wait for another test from somebody else or can we merge it ?

**Pull request merged**